tools/make.py

```diff
@@ -38,157 +38,153 @@
 from tools.paths import UBLOX_LIBRARY
 from tools.tests import TESTS, Test, TEST_MAP
 from tools.tests import TEST_MBED_LIB
+from tools.tests import test_known, test_name_known
 from tools.targets import TARGET_MAP
 from tools.options import get_default_options_parser
 from tools.build_api import build_project
 from tools.build_api import mcu_toolchain_matrix
-try:
-    import tools.private_settings as ps
-except:
-    ps = object()
+from utils import argparse_filestring_type
+from argparse import ArgumentTypeError
 
 if __name__ == '__main__':
     # Parse Options
     parser = get_default_options_parser()
-    parser.add_option("-p",
-                      type="int",
+    group = parser.add_mutually_exclusive_group(required=True)
```

Comment:
This `required` causes an issue when the `-L <--list-tests>` option is specified:

```
$ python tools\make.py -L
usage: make.py [-h] [-m MCU] [-t TOOLCHAIN] [-c] [-o OPTIONS]
               (-p PROGRAM | -n PROGRAM) [-j JOBS] [-v] [--silent] [-D MACROS]
               [-S] [-f GENERAL_FILTER_REGEX] [--automated] [--host HOST_TEST]
               [--extra EXTRA] [--peripherals PERIPHERALS]
               [--dep DEPENDENCIES] [--source SOURCE_DIR]
               [--duration DURATION] [--build BUILD_DIR] [-N ARTIFACT_NAME]
               [-d DISK] [-s SERIAL] [-b BAUD] [-L] [--rtos] [--rpc] [--eth]
               [--usb_host] [--usb] [--dsp] [--fat] [--ublox] [--testlib]
               [-l LINKER_SCRIPT]
make.py: error: one of the arguments -p -n is required
```